src/agent.cpp

```diff
@@ -62,6 +62,31 @@ namespace Rep
 
     std::string Agent::escape(const std::string& query)
     {
-        return Url::Url(query).defrag().escape().fullpath();
+        // Special case: treat multiple leading slashes as one.
+        std::string query_;
+        if (query.size() > 1 && query[0] == '/')
+        {
+            query_.reserve(query.size());
+            query_.append(1, '/');
+            bool leading_slashes_consumed = false;
+            for (auto character : query)
```

Comment:
Personally, I would probably not copy a single character at a time. This type of loop is not very friendly to the optimizer since there's an `if` in the middle of it. I would do something like this:
```c++
char* start = &query[0];
while (*start == '/') {
    ++start;
}
query_.assign(start, &query[query.size()]);
```

Also, since a leading slash in the query is a common case, it might be better to only do the copy when there's at least two leading slashes:
```c++
if (query.size() >= 2 && query[0] == '/' && query[1] == '/')
{
    query_.reserve(query.size());
    query_.append(1, '/');
    // Starting at 2 since we know the first two characters are both already '/'
    char* start = &query[2];
    char* end = &query[query.size()];
    while (start != end && *start == '/')
    {
        ++start;
    }
    query_.assign(start, end);
}
```